opa_client/src/lib.rs

```diff
@@ -0,0 +1,85 @@
+#[cfg(feature = "graphql")]
+pub mod graphql;
+
+use derive_more::Deref;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use url::Url;
+
+#[derive(Debug, thiserror::Error)]
+pub enum OPADecisionError {
+    #[error("Could not create request URL")]
+    InvalidPath(#[from] url::ParseError),
+    #[error("Request to OPA server failed")]
+    RequestFailed(#[from] reqwest::Error),
+}
+
+#[derive(Debug, Clone)]
+pub struct OPAClient {
+    url: Url,
+}
+
+#[derive(Debug, Serialize)]
+struct OPARequest<T> {
+    input: T,
+}
```

Comment:
Do we want any client-side validation? Currently I can make up any struct I want and send it to OPA, whether OPA will recognise it or not. It might nice if this fails to built if someone tries to send it nonsense.

Reply:
That'd certainly be nice to have, unfortunately the endpoint descriptors exposed by OPA are non-standard, so will take quite a bit of work. Definitely a goal in the future if we settle on OPA for AuthZ